ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLOperationContext.java

```diff
@@ -51,8 +51,8 @@ public DDLOperationContext(HiveConf conf, Context context, DDLTask task, DDLWork
     this.console = console;
   }
 
-  public Hive getDb() {
-    return db;
+  public Hive getDb() throws HiveException {
+    return Hive.getCurrHiveDb(db, conf);
```

Comment:
Is there a situation when this is `null` or `closed`?

Reply:
Yes, there is a situation where hive connections is closed and it still returns reference of the closed hive object.